assert/assertions.go

```diff
@@ -2292,3 +2301,357 @@ func buildErrorChainString(err error, withType bool) string {
+// It provides detailed error messages when differences are found, including the specific fields that differ.
+func inspectSlices(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}) bool {
+	expectedValue := reflect.ValueOf(expected)
+	actualValue := reflect.ValueOf(actual)
+
+	expectedLen := expectedValue.Len()
+	actualLen := actualValue.Len()
+
+	minLen := expectedLen
+	if actualLen < minLen {
+		minLen = actualLen
+	}
+
+	var differences []string
+
+	if actualLen != expectedLen {
+		Fail(t, fmt.Sprintf("Lengths not equal:\nexpected: %d\nactual  : %d", expectedLen, actualLen), msgAndArgs...)
+		return false
+	}
+
+	for i := 0; i < minLen; i++ {
+		expectedItem := expectedValue.Index(i).Interface()
+		actualItem := actualValue.Index(i).Interface()
+
+		if !ObjectsAreEqual(expectedItem, actualItem) {
+			message := messageFromMsgAndArgs(msgAndArgs...)
+			diffs := findDifferences(expectedItem, actualItem)
+
+			differencesOutput := "Field differences:\n"
+			for _, diff := range diffs {
+				differencesOutput += fmt.Sprintf("  └─ %s: %s ≠ %s\n", diff.Path, formatDiffValue(diff.Expected), formatDiffValue(diff.Actual))
+			}
+
+			if message != "" {
+				differences = append(differences, fmt.Sprintf(
+					"MATCH: %s\n[index %d] Not equal:\nexpected: %v\nactual  : %v",
+					message, i, formatComparisonValue(expectedItem), formatComparisonValue(actualItem)))
+				differences = append(differences, differencesOutput)
+			} else {
+				differences = append(differences, fmt.Sprintf(
+					"MATCH: [index %d] Not equal:\nexpected: %v\nactual  : %v",
+					i, formatComparisonValue(expectedItem), formatComparisonValue(actualItem)))
+				differences = append(differences, differencesOutput)
+			}
+		}
+	}
+
+	if len(differences) > 0 {
+		diffMessage := strings.Join(differences, "\n")
+		Fail(t, fmt.Sprintf("Differences found:\n%s", diffMessage), msgAndArgs...)
+		return false
+	}
+
+	return true
+}
+
+// formatComparisonValue formats an arbitrary value for human-readable comparison.
+// It applies proper formatting based on the value's type.
+func formatComparisonValue(obj interface{}) string {
+	return formatValueComparison(reflect.ValueOf(obj))
+}
+
+// formatValueComparison handles the formatting logic for different reflect.Value types
+// to provide consistent and readable output for comparison purposes.
+func formatValueComparison(v reflect.Value) string {
+	switch v.Kind() {
+	case reflect.Struct:
+		var parts []string
+		t := v.Type()
+		for i := 0; i < v.NumField(); i++ {
+			field := t.Field(i)
+			if !field.IsExported() {
+				continue
+			}
+			fieldValue := v.Field(i)
+			parts = append(parts, fmt.Sprintf("%s: %s", field.Name, formatValueComparison(fieldValue)))
+		}
+		return fmt.Sprintf("{%s}", strings.Join(parts, ", "))
+
+	case reflect.String:
+		return fmt.Sprintf(`"%s"`, v.String())
+
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return fmt.Sprintf("%d", v.Int())
```

Comment:
I feel like most of the Sprintf could be this, because they are almost equivalent to sprintf("%v", whatever) so sprint(whatever)

```suggestion
		return fmt.Sprint(v.Int())
```

Please double check.

You can use perfsprint linter if you are curious